doi_downloader/plugins/unpaywall.py

```diff
@@ -34,23 +38,40 @@ def fetch_metadata(self, doi):
             return None
 
     # Function to get the URL of the PDF from the DOI
-    def get_pdf_url(self, doi, use_cache=True, ttl=0):
+    def get_pdf_url(self, doi, ctx, use_cache=True, ttl=0, enable_benchmark=False):
```

Comment:
Make new argument `ctx` optional to keep code backward compatible: `get_pdf_url(self, doi, ctx=None, use_cache=True, ttl=0, enable_benchmark=False)`